src/cephadm/cephadm

```diff
@@ -2990,9 +2990,18 @@ def command_logs():
     if not args.fsid:
         raise Error('must pass --fsid to specify cluster')
 
-    (daemon_type, daemon_id) = args.name.split('.', 1)
+    try:
+        (daemon_type, daemon_id) = args.name.split('.', 1)
+    except ValueError:
+        raise Error('--name: daemon id required. Something like <type>.<id>')
+
     unit_name = get_unit_name(args.fsid, daemon_type, daemon_id)
 
+    enabled, status, installed = check_unit(unit_name)
+    if not installed:
+        raise Error(f"unknown daemon. See `cephadm ls`")
```

Comment:
unfortunately, this doesn't completely work:

```
$ systemctl is-enabled ceph-376df762-3ba4-42eb-9ec9-795b9b4d11c8@blablablablaunknown
disabled
```

any ideas?

Reply:
What output would you expect?

Reply:
```
Failed to get unit file state for ceph-376df762-3ba4-42eb-9ec9-795b9b4d11c8@blablablablaunknown: No such file or directory
```

Reply:
add a new list func for the query?

```
diff --git a/src/cephadm/cephadm b/src/cephadm/cephadm
index 7c5d2815f8..9140580e4f 100755
--- a/src/cephadm/cephadm
+++ b/src/cephadm/cephadm
@@ -2990,17 +2990,13 @@ def command_logs():
     if not args.fsid:
         raise Error('must pass --fsid to specify cluster')
 
-    try:
-        (daemon_type, daemon_id) = args.name.split('.', 1)
-    except ValueError:
-        raise Error('--name: daemon id required. Something like <type>.<id>')
+    d = list_daemon(args.fsid, args.name)
+    if not d:
+        raise Error('Unknown daemon: %s' % (args.name))
 
+    (daemon_type, daemon_id) = args.name.split('.', 1)
     unit_name = get_unit_name(args.fsid, daemon_type, daemon_id)
 
-    enabled, status, installed = check_unit(unit_name)
-    if not installed:
-        raise Error(f"unknown daemon. See `cephadm ls`")
-
     cmd = [find_program('journalctl')]
     cmd.extend(['-u', unit_name])
     if args.command:
@@ -3052,6 +3048,16 @@ def command_ls():
                       legacy_dir=args.legacy_dir)
     print(json.dumps(ls, indent=4))
 
+def list_daemon(fsid, name, detail=False, legacy_dir=None):
+    # type: (str, str, bool, Optional[str]) -> Dict[str, str]
+    for d in list_daemons(detail=detail, legacy_dir=legacy_dir):
+        if d['fsid'] != fsid:
+            continue
+        if d['name'] != name:
+            continue
+        return d
+    return {}
+
 def list_daemons(detail=True, legacy_dir=None):
     # type: (bool, Optional[str]) -> List[Dict[str, str]]
     host_version = None

```

Reply:
you could also check `systemctl status` for a exitcode of 4 .. but I think adding a new list_daemon (or similar) func would more flexible ..

Reply:
fixed!